syntax = "proto3";

import public "proto/data.proto";

message NotificationRequest {
    string email = 1;
    repeated Course classes = 2;
}

message ScheduleRequest {
    Major major = 1;
    string semester = 2;
    int32 minCredits = 3;
    int32 maxCredits = 4;
    repeated Course previousClasses = 5;
    repeated Course preferredClasses = 6;
    repeated Course unpreferredClasses = 7;
    repeated Professor preferredProfs = 8;
    repeated Professor unprefferedProfs = 9;
}

message MajorRequest {}

message ProfessorRequest {}

message CourseRequest {}

message DebugRequest {
    string msg = 1;
}
